### PR TITLE
backend-notification-pusher: Dynamically discover remote domains

### PR DESCRIPTION
**Pull request by akshaymankar**
https://wearezeta.atlassian.net/browse/WPB-1130

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: libs/extended/src/Network/AMQP/Extended.hs [hunk 1]
```diff
@@ -119,3 +182,9 @@ logException l m (SomeException e) = do
   Log.err l $
     Log.msg m
       . Log.field "error" (displayException e)
+
+readCredsFromEnv :: IO (Text, Text)
+readCredsFromEnv = do
+  username <- Text.pack <$> getEnv "RABBITMQ_USERNAME"
+  password <- Text.pack <$> getEnv "RABBITMQ_PASSWORD"
+  pure (username, password)
```
Comment (battermann) [inline libs/extended/src/Network/AMQP/Extended.hs:190]: I like applicative syntax, but that is a matter of taste.

```suggestion
readCredsFromEnv =
  (,) <$> (Text.pack <$> getEnv "RABBITMQ_USERNAME") <*> (Text.pack <$> getEnv "RABBITMQ_PASSWORD")
```

File: services/background-worker/test/Test/Wire/BackendNotificationPusherSpec.hs [hunk 2]
```diff
@@ -19,122 +40,280 @@ import Wire.API.Federation.BackendNotifications
...
 import Wire.BackgroundWorker.Env
+import Wire.BackgroundWorker.Options
 
 runTestAppT :: AppT IO a -> Int -> IO a
 runTestAppT app port = do
   http2Manager <- initHttp2Manager
   logger <- Logger.new Logger.defSettings
   let federatorInternal = Endpoint "localhost" (fromIntegral port)
+      rabbitmqAdminClient = undefined
+      rabbitmqVHost = undefined
```
Comment (battermann) [inline services/background-worker/test/Test/Wire/BackendNotificationPusherSpec.hs:51]: I guess `undefined` is fine here?

File: services/background-worker/test/Test/Wire/BackendNotificationPusherSpec.hs [hunk 3]
```diff
@@ -19,122 +40,280 @@ import Wire.API.Federation.BackendNotifications
...
+          False
+          [ "backend-notifications.foo.example",
+            "backend-notifications.bar.example",
+            "some-other-queue",
+            "backend-notifications.baz.example",
+            "backend-notifications.not-a-domain"
+          ]
+      logger <- Logger.new Logger.defSettings
+      let federatorInternal = undefined
+          http2Manager = undefined
```
Comment (battermann) [inline services/background-worker/test/Test/Wire/BackendNotificationPusherSpec.hs:166]: I guess this is fine, too?

**Review by battermann — APPROVED** [review]

File: services/background-worker/test/Test/Wire/BackendNotificationPusherSpec.hs (hunk 2 above)
Comment (akshaymankar) [reply services/background-worker/test/Test/Wire/BackendNotificationPusherSpec.hs:51]: Yeah, I am trying to write tests without polysemy, I think this is the easiest thing we can do.

File: services/background-worker/test/Test/Wire/BackendNotificationPusherSpec.hs (hunk 3 above)
Comment (akshaymankar) [reply services/background-worker/test/Test/Wire/BackendNotificationPusherSpec.hs:166]: Same as above.

File: libs/extended/src/Network/AMQP/Extended.hs (hunk 1 above)
Comment (akshaymankar) [reply libs/extended/src/Network/AMQP/Extended.hs]: I added some new lines. Its applicatives all the way down :turtle:

**Pull request merged**